OpenRA.Mods.Common/Traits/Render/WithDamageOverlay.cs

```diff
@@ -70,16 +93,21 @@ void INotifyDamage.Damaged(Actor self, AttackInfo e)
 			if (!info.DamageTypes.IsEmpty && !e.Damage.DamageTypes.Overlaps(info.DamageTypes))
 				return;
 
-			if (isSmoking) return;
+			if (isPlayingAnimation) return;
 			if (e.Damage.Value < 0) return; /* getting healed */
 			if (e.DamageState < info.MinimumDamageState) return;
 			if (e.DamageState > info.MaximumDamageState) return;
 
-			isSmoking = true;
-			anim.PlayThen(info.IdleSequence,
-				() => anim.PlayThen(info.LoopSequence,
-					() => anim.PlayThen(info.EndSequence,
-						() => isSmoking = false)));
+			isPlayingAnimation = true;
+			PlayAnimation(info.GetSequences(self.World.LocalRandom).GetEnumerator());
+		}
```

Comment:
So diff players could see the same tree burn a different time length?

Won't cause false positive issue/bug reports?

Why not just use the network random?

Reply:
I don't think it's an issue. Visuals are allowed to not stay synced